src/main/scala/com/databricks/spark/redshift/RedshiftRelation.scala

```diff
@@ -0,0 +1,131 @@
+/*
+ * Copyright 2015 TouchType Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.databricks.spark.redshift
+
+import java.util.Properties
+
+import com.databricks.spark.redshift.Parameters.MergedParameters
+import org.apache.spark.Logging
+import org.apache.spark.rdd.RDD
+import org.apache.spark.sql.jdbc.JDBCWrapper
+import org.apache.spark.sql.sources._
+import org.apache.spark.sql.types._
+import org.apache.spark.sql.{DataFrame, Row, SQLContext}
+
+/**
+ * Data Source API implementation for Amazon Redshift database tables
+ */
+private [redshift]
+case class RedshiftRelation(jdbcWrapper: JDBCWrapper,
+                            params: MergedParameters,
+                            userSchema: Option[StructType])
+                           (@transient val sqlContext: SQLContext)
+  extends BaseRelation
+  with TableScan
+  with PrunedScan
+  with PrunedFilteredScan
+  with InsertableRelation
+  with Logging {
+
+  override def schema = {
+    userSchema match {
+      case Some(schema) => schema
+      case None => {
+        jdbcWrapper.registerDriver(params.jdbcDriver)
+        jdbcWrapper.resolveTable(params.jdbcUrl, params.table, new Properties())
+      }
+    }
+  }
+
+  override def buildScan(): RDD[Row] = {
+    unloadToTemp()
+    makeRdd(schema)
+  }
+
+  override def buildScan(requiredColumns: Array[String]): RDD[Row] = {
+    val columns = columnList(requiredColumns)
+    unloadToTemp(columns)
+    makeRdd(pruneSchema(schema, requiredColumns))
+  }
+
+  override def buildScan(requiredColumns: Array[String], filters: Array[Filter]): RDD[Row] = {
```

Comment:
You only need to implement this one (and you can remove the other Scan traits as they will never be called)